enclave/crypto/gcm.c

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <mbedtls/cipher.h>
+
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/crypto/gcm.h>
+
+oe_result_t oe_aes_gcm_encrypt(
+    const uint8_t* key,
+    size_t key_size,
+    const uint8_t* iv,
+    size_t iv_size,
+    const uint8_t* aad,
+    size_t aad_size,
+    const uint8_t* input,
+    size_t inlen,
+    uint8_t* output,
+    uint8_t* tag)
+{
+    const mbedtls_cipher_info_t *info;
+    mbedtls_cipher_context_t gcm;
+    oe_result_t result = OE_OK;
+    size_t olen;
+
+    if (key_size != 16)
```

Comment:
`mbedtls_cipher_definitions[MBEDTLS_CIPHER_AES_128_GCM].key_bitlen` would be a more descriptive "the default key size for this cipher" which I think is the intention here?

Reply:
Fixed. Please see the latest commit.